storage/fsStorage.go

```diff
@@ -1,20 +1,84 @@
 package storage
 
-import "fmt"
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+)
 
-type FsStorage struct {
+type fsStorage struct {
 	path string
 }
+type myStruct struct {
+	Key   string
+	Value string
+}
+
+func (st fsStorage) Read(key string) (string, error) {
+	allDataFromJson := []myStruct{}
+	file, err := ioutil.ReadFile(st.path)
+	if err != nil {
+		return "", err
+	}
+	json.Unmarshal(file, &allDataFromJson)
+	if err != nil {
+		return "", err
+	}
+	for _, v := range allDataFromJson {
+		if v.Key == key {
+			return v.Value, nil
+		}
+	}
+	return "", errors.New("item was not found")
+}
+func (st fsStorage) Write(key string, value string) error {
+	err := checkFile(st.path)
+	if err != nil {
+		return err
+	}
+
+	file, err := ioutil.ReadFile(st.path)
+	if err != nil {
+		return err
+	}
 
-func (st FsStorage) Read(key string) (string, error) {
-	return "got it", nil
+	data := []myStruct{}
+
+	json.Unmarshal(file, &data)
+
+	newStruct := &myStruct{
+		Key:   key,
+		Value: value,
+	}
+
+	data = append(data, *newStruct)
+	dataBytes, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(st.path, dataBytes, 0644)
+	if err != nil {
+		return err
+	}
+	return nil
 }
-func (st FsStorage) Write(key string, value string) {
-	fmt.Print(value)
+
+func NewFsStorage() fsStorage {
+	return fsStorage{
+		path: "test.json",
```

Comment:
Pass filename as constructor's argument.

Reply:
done in the[ last commit ](https://github.com/retttddd/golangProjectv3/pull/1/commits/c0c25b2a0ea0e478cf0e24869c8800e78b375e11)